src/Gui/SoDatumLabel.cpp

```diff
@@ -1030,3 +1036,265 @@ void SoDatumLabel::setPoints(SbVec3f p1, SbVec3f p2)
     verts[1] = p2;
     pnts.finishEditing();
 }
+
+
+struct NodeData {
+    EditableDatumLabel* label;
+};
+
+/*********************** EditableDatumLabel ***************************************/
+
+EditableDatumLabel::EditableDatumLabel(View3DInventorViewer* view, Base::Placement plc, SbColor color, bool autoDistance)
+    : viewer(view)
+    , spinBox(nullptr)
+    , cameraSensor(nullptr)
+    , isSet(false)
+    , autoDistance(autoDistance)
+    , autoDistanceReverse(false)
+    , value(0.)
+{
+    root = new SoAnnotation;
+    root->ref();
+    root->renderCaching = SoSeparator::OFF;
+
+    transform = new SoTransform();
+    root->addChild(transform);
+
+    label = new SoDatumLabel();
+    label->ref();
+    label->string = " ";
+    label->textColor = color;
+    label->size.setValue(17);
+    label->lineWidth = 2.0;
+    label->useAntialiasing = false;
+    label->datumtype = SoDatumLabel::DISTANCE;
+    label->param1 = 0.;
+    label->param2 = 0.;
+    if (autoDistance)
+        setLabelRecommendedDistance();
+    root->addChild(label);
+
+    setPlacement(plc);
+}
+
+EditableDatumLabel::~EditableDatumLabel()
+{
+    deactivate();
+    transform->unref();
+    root->unref();
```

Comment:
When adding `transform->unref()` you have to do `transform->ref();` in the constructor as otherwise the application will segfault.